src/libMessage/Messenger.cpp

```diff
@@ -4740,3 +4741,207 @@ bool Messenger::GetConsensusCommitFailure(
+  LookupSetDirectoryBlocksFromSeed result;
+
+  result.set_indexnum(index_num);
+
+  for (const auto& dirblock : directoryBlocks) {
+    ProtoSingleDirectoryBlock* proto_dir_blocks = result.add_dirblocks();
+    if (dirblock.type() == typeid(DSBlock)) {
+      DSBlockToProtobuf(get<DSBlock>(dirblock),
+                        *proto_dir_blocks->mutable_dsblock());
+    } else if (dirblock.type() == typeid(VCBlock)) {
+      VCBlockToProtobuf(get<VCBlock>(dirblock),
+                        *proto_dir_blocks->mutable_vcblock());
+    } else if (dirblock.type() == typeid(FallbackBlockWShardingStructure)) {
+      FallbackBlockToProtobuf(
+          get<FallbackBlockWShardingStructure>(dirblock).m_fallbackblock,
+          *proto_dir_blocks->mutable_fallbackblockwshard()
+               ->mutable_fallbackblock());
+      ShardingStructureToProtobuf(
+          get<FallbackBlockWShardingStructure>(dirblock).m_shards,
+          *proto_dir_blocks->mutable_fallbackblockwshard()->mutable_sharding());
+    }
+  }
+
+  if (!result.IsInitialized()) {
+    LOG_GENERAL(WARNING,
+                "LookupSetDirectoryBlocksFromSeed initialization failed");
+  }
+
+  return SerializeToArray(result, dst, offset);
+}
+
+bool Messenger::GetLookupSetDirectoryBlocksFromSeed(
+    const vector<unsigned char>& src, const unsigned int offset,
+    vector<boost::variant<DSBlock, VCBlock, FallbackBlockWShardingStructure>>&
+        directoryBlocks,
+    uint64_t& index_num) {
+  LookupSetDirectoryBlocksFromSeed result;
+
+  result.ParseFromArray(src.data() + offset, src.size() - offset);
+
+  if (!result.IsInitialized()) {
+    LOG_GENERAL(WARNING,
+                "LookupSetDirectoryBlocksFromSeed initialization failed");
+    return false;
+  }
+
+  index_num = result.indexnum();
+
+  for (const auto& dirblock : result.dirblocks()) {
+    switch (dirblock.directoryblock_case()) {
+      case ProtoSingleDirectoryBlock::DirectoryblockCase::kDsblock: {
+        DSBlock dsblock;
+        ProtobufToDSBlock(dirblock.dsblock(), dsblock);
+        directoryBlocks.emplace_back(dsblock);
+        break;
+      }
+      case ProtoSingleDirectoryBlock::DirectoryblockCase::kVcblock: {
+        VCBlock vcblock;
+        ProtobufToVCBlock(dirblock.vcblock(), vcblock);
+        directoryBlocks.emplace_back(vcblock);
+        break;
+      }
+      case ProtoSingleDirectoryBlock::DirectoryblockCase::
+          kFallbackblockwshard: {
+        FallbackBlockWShardingStructure fallbackblockwshard;
+        ProtobufToFallbackBlock(dirblock.fallbackblockwshard().fallbackblock(),
+                                fallbackblockwshard.m_fallbackblock);
+        ProtobufToShardingStructure(dirblock.fallbackblockwshard().sharding(),
+                                    fallbackblockwshard.m_shards);
+        directoryBlocks.emplace_back(fallbackblockwshard);
+        break;
+      }
+      default: {
+        LOG_GENERAL(WARNING, "Error in the blocktype");
+        continue;
```

Comment:
I think common coding convention uses `break` even for the `default` case, and also doesn't use curly braces for each switch case. Unless, our clang added those braces by default?

Reply:
I think I added the curly braces. noted.